scripts/commit_classifier.py

```diff
@@ -582,25 +592,12 @@ def feature_sort_key(f):
     def classify(
         self,
         revision: Optional[str] = None,
-        phabricator_deployment: Optional[str] = None,
-        diff_id: Optional[int] = None,
         runnable_jobs_path: Optional[str] = None,
     ) -> None:
-        if revision is not None:
-            assert phabricator_deployment is None
-            assert diff_id is None
-
-        if diff_id is not None:
-            assert phabricator_deployment is not None
-            assert revision is None
-
         self.update_commit_db()
 
-        if phabricator_deployment is not None and diff_id is not None:
-            with hglib.open(self.repo_dir) as hg:
-                self.apply_phab(hg, phabricator_deployment, diff_id)
-
-                revision = hg.log(revrange="not public()")[0].node.decode("utf-8")
+        if self.revision is not None:
+            revision = self.revision
```

Comment:
I would add `assert revision is None`  here in case the method gets called outside the main with the wrong args.

Reply:
Ok sir, noted